src/components/carousel/index.jsx

```diff
@@ -90,20 +101,38 @@ const Carousel = ({
 		Children.map(children, (child) => (child?.type === subcomponentType ? child : null))
 	);
 
+	useEffect(() => {
+		setSlidesToShowInView(getSlidesToShowFromDom(id));
+		setSlide(getSlidesToShowFromDom(id));
+	}, [id]);
+
+	useEffect(() => {
+		const resizeFn = () => {
+			if (slidesToShowInView === 0 || getSlidesToShowFromDom(id) === slidesToShowInView) {
+				return;
+			}
+			setSlidesToShowInView(getSlidesToShowFromDom(id));
+			setSlide(getSlidesToShowFromDom(id));
+			setPosition(0);
+		};
+		window.addEventListener("resize", resizeFn, false);
+		return () => window.addEventListener("resize", resizeFn, false);
```

Comment:
Assuming you meant to remove the event listener here.

Reply:
Update